torch/_inductor/codegen/memory_planning.py

```diff
@@ -1,3 +1,5 @@
+from __future__ import annotations
```

Comment:
I had to import this because older Python runtimes don't seem happy when encountering `CachedMethod[[], Any]`; they don't like `[]` being passed as the value of a `ParamSpec` because they treat it as a regular `Generic` typevar. This import makes it so that type hints don't get automatically evaluated at runtime.

This import then made `ruff` un-quote the string type hints in the rest of the file, hence the numerous changes

Reply:
@int3 Just want to confirm, using the backported version of typing_extensions.Protocol: https://typing-extensions.readthedocs.io/en/latest/#Protocol does not solve the issue. (Although it appears like it introduces other ones, sadly).

Reply:
Hm yeah I don't see how `Protocol` would solve things here...